### DISCUSSION TITLE
Can I import private key/seedphrase from another wallet into web3auth

### DISCUSSION THREAD

**Discussion by nakulmk** [Web3Auth/Web3Auth#336] [Help]
Can I import a private key/seedphrase from another wallet into web3auth?

**Comment by YZhenY** [comment]
Imports have always been tricky with key management solutions, as it inherits the security/insecurity of the previous way the key was managed. Thus we currently support imports, but only with our Self-Hosted Web3Auth solution (or tkey).

With Web3Auth's plug and play its some thing we're thinking about but its not currently supported. Feel free to create a feature request as we'd love to gauge demand

**Reply by sandys** [reply]
@YZhenY is there an example of this ? with self-hosted solution. we are looking for this as well, where we would like to get users to bring their wallets into web3auth and keep them there.

**Reply by YZhenY** [reply]
@sandys take a look here https://web3auth.io/docs/guides/selfhost
